File: backend/app/schemas/resume.py
from pydantic import BaseModel, Field


class ResumeSkill(BaseModel):
    skill: str
    reason: str | None = None


class ResumeSoftSkill(BaseModel):
    text: str
    evidence: str | None = None


class ResumeProject(BaseModel):
    name: str
    description: str | None = None
    tech_stack: list[ResumeSkill] = Field(default_factory=list)


class ResumeStrength(BaseModel):
    text: str
    evidence: str


class ResumeWeakness(BaseModel):
    text: str
    evidence: str


class ResumeMissingSkill(BaseModel):
    skill: str
    reason: str


class RecommendedTopic(BaseModel):
    topic: str
    reason: str


class ResumeAnalysisResponse(BaseModel):
    resume_score: int = Field(..., ge=0, le=100)

    technical_skills: list[ResumeSkill] = Field(default_factory=list)
    soft_skills: list[ResumeSoftSkill] = Field(default_factory=list)

    projects: list[ResumeProject] = Field(default_factory=list)

    strengths: list[ResumeStrength] = Field(default_factory=list)
    weaknesses: list[ResumeWeakness] = Field(default_factory=list)
    missing_skills: list[ResumeMissingSkill] = Field(default_factory=list)

    recommended_topics: list[RecommendedTopic] = Field(default_factory=list)

File: frontend/src/types/resume.ts
export interface ResumeSkill {
  skill: string;
  reason: string | null;
}

export interface ResumeSoftSkill {
  text: string;
  evidence: string | null;
}

export interface ResumeProject {
  name: string;
  description: string | null;
  tech_stack: ResumeSkill[];
}

export interface ResumeStrength {
  text: string;
  evidence: string;
}

export interface ResumeWeakness {
  text: string;
  evidence: string;
}

export interface ResumeMissingSkill {
  skill: string;
  reason: string;
}

export interface RecommendedTopic {
  topic: string;
  reason: string;
}

export interface ResumeAnalysis {
  resume_score: number;

  technical_skills: ResumeSkill[];

  soft_skills: ResumeSoftSkill[];

  projects: ResumeProject[];

  strengths: ResumeStrength[];

  weaknesses: ResumeWeakness[];

  missing_skills: ResumeMissingSkill[];

  recommended_topics: RecommendedTopic[];
}

export interface ResumeHistoryItem {
  id: number;
  filename: string;
  resume_score: number;
  created_at: string;
}

export interface ResumeHistoryResponse {
  items: ResumeHistoryItem[];
}